site/cds_rdm/tasks.py

```diff
@@ -140,3 +147,224 @@ def sync_users():
+
+    def _update_name(user, name_dict, is_orcid=False):
+        """Updates the name with the new values."""
+        updated_name = _check_if_update_needed(user, name_dict, is_orcid)
+        if updated_name:
+            try:
+                service.update(system_identity, name_dict.get("id"), updated_name)
+            except ValidationError as e:
+                current_app.logger.error(f"Error updating name for user {user.id}: {e}")
+
+    def _create_new_name(user):
+        name = {"id": str(user.id), "props": {"is_cern": True}}
+
+        if user.user_profile.get("given_name"):
+            name["given_name"] = user.user_profile.get("given_name")
+        if user.user_profile.get("family_name"):
+            name["family_name"] = user.user_profile.get("family_name")
+        if user.user_profile.get("affiliations"):
+            name["affiliations"] = [{"name": user.user_profile.get("affiliations", "")}]
+        if user.user_profile.get("orcid"):
+            name["identifiers"] = [
+                {"scheme": "orcid", "identifier": user.user_profile.get("orcid")}
+            ]
+        for prop in prop_values:
+            if user.user_profile.get(prop):
+                name["props"][prop] = user.user_profile.get(prop)
+        try:
+            service.create(system_identity, name)
+        except ValidationError as e:
+            current_app.logger.error(f"Error creating name for user {user.id}: {e}")
+
+    users = []
+    # Allows to sync a single user
+    if user_id:
+        user = User.query.get(user_id)
+        users = [user] if user else []
+    else:
+        users = User.query.filter(User.updated > since).all()
+    name_dict = None
+    for user in users:
+        orcid = None
+        try:
+            name_dict = service.read(system_identity, str(user.id)).to_dict()
+            _update_name(user, name_dict)
+        except NoResultFound:
+            orcid = user.user_profile.get("orcid")
+            if orcid:
+                try:
+                    # Check if the ORCID value is already present and update it
+                    name_dict = service.read(system_identity, orcid).to_dict()
+                    _update_name(user, name_dict, is_orcid=True)
+                except NoResultFound:
+                    _create_new_name(user)
+            else:
+                _create_new_name(user)
+
+
+
+@shared_task()
+def merge_duplicate_names_vocabulary(since=None):
+    """Merges duplicate names in the names vocabulary."""
+    service = current_service_registry.get("names")
+
+    def _merge(name_from, name_to):
+        """Merges the names.
+
+        param name_from: The name to merge from. This will marked as unlisted after the merge, meaning that won't be returned in the search results.
+        param name_to: The name to merge to.
+        """
+        updated = False
+        # Merge the affiliations
+        affiliations_to = name_to.get("affiliations", [])
```

Comment:
it would be a little bit less "tangled" if we call it `affiliations_source` instead of `affiliations_from`  and `affiliations_dest`, but lets hear what the rest of the team thinks before changing it